vulture_os/toolkit/api_parser/beyondtrust_reportings/beyondtrust_reportings.py

```diff
@@ -0,0 +1,325 @@
+
+    def remove_hash_from_keys(self, d):
+        if isinstance(d, dict):
+            return {key.replace('#', ''): self.remove_hash_from_keys(value) for key, value in d.items()}
+        elif isinstance(d, list):
+            return [self.remove_hash_from_keys(item) for item in d]
+        else:
+            return d
+
+    def format_log(self, log, resource, requested_type):
+        log["resource"] = resource
+        log["requested_type"] = requested_type
+        cleaned_log = self.remove_hash_from_keys(log)
+        return json_dumps(cleaned_log)
+
+    def execute(self):
+
+        # Get reporting logs
+        reporting_types = [r_type for r_type, value in self.allowed_reporting_types.items() if value]
+
+        if not reporting_types:
+            msg = f"[{__parser__}]:execute: Reporting types list cannot be empty."
+            raise BeyondtrustReportingsAPIError(msg)
+
+        for report_type in reporting_types:
+            since = self.last_collected_timestamps.get(f"beyondtrust_reportings_{report_type}") or (timezone.now() - timedelta(days=30))
+
+            msg = f"Parser starting to get {report_type} logs from {since}"
+            logger.info(f"[{__parser__}]:execute: {msg}", extra={'frontend': str(self.frontend)})
+
+            while since < timezone.now() - timedelta(hours=1) and not self.evt_stop.is_set():
+
+                logs, last_datetime = self.get_logs("reporting", report_type, since)
+
+                if logs:
+                    self.write_to_file([self.format_log(log, "reporting", report_type) for log in logs])
+                    # Downloading may take some while, so refresh token in Redis
+                    self.update_lock()
+
+                try:
+                    last_datetime = datetime.fromtimestamp(int(last_datetime), tz=timezone.utc) + timedelta(seconds=1)
+                    self.last_collected_timestamps[f"beyondtrust_reportings_{report_type}"] = last_datetime
+                except (TypeError, ValueError):
+                    # Update timestamp +24 if since < now, otherwise +1
+                    delta = 24 if since < timezone.now() - timedelta(hours=24) else 1
+                    msg = f"No logs, advancing timestamp by {delta} hour(s)"
+                    logger.debug(f"[{__parser__}]:execute: {msg}", extra={'frontend': str(self.frontend)})
+                    self.last_collected_timestamps[f"beyondtrust_reportings_{report_type}"] = since + timedelta(hours=delta)
+                finally:
+                    msg = "Current timestamp for {report_type} is {timestamp}".format(
+                        report_type=report_type,
+                        timestamp=self.last_collected_timestamps[f"beyondtrust_reportings_{report_type}"]
+                    )
+                    logger.debug(f"[{__parser__}]:execute: {msg}", extra={'frontend': str(self.frontend)})
+                    since = self.last_collected_timestamps[f"beyondtrust_reportings_{report_type}"]
```

Comment:
```suggestion
                    logger.info(f"[{__parser__}]:execute: {msg}", extra={'frontend': str(self.frontend)})
                    self.last_collected_timestamps[f"beyondtrust_reportings_{report_type}"] = since + timedelta(hours=delta)
                finally:
                    msg = "New timestamp for {report_type} is {timestamp}".format(
                        report_type=report_type,
                        timestamp=self.last_collected_timestamps[f"beyondtrust_reportings_{report_type}"]
                    )
                    logger.info(f"[{__parser__}]:execute: {msg}", extra={'frontend': str(self.frontend)})
                    since = self.last_collected_timestamps[f"beyondtrust_reportings_{report_type}"]
```